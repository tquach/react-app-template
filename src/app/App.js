import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';

import '../styles/style.scss';

class App extends Component {
  render() {
    return (
      <main data-theme="default">
        <div className="container container--full-width">
          <div className="nav" />
        </div>
        <Switch>
          <Route
            exact
            path="/"
            component={() => {
              return 'Hello, world!';
            }}
          />
        </Switch>
      </main>
    );
  }
}

export default App;
