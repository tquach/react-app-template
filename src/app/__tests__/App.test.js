import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';


import App from '../App';

describe('App', () => {
  beforeAll(() => {
    ReactDOM.createPortal = jest.fn(element => {
      return element;
    });
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <App />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
