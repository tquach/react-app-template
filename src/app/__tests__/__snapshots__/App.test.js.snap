// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<main
  data-theme="default"
>
  <div
    className="container container--full-width"
  >
    <div
      className="nav"
    />
  </div>
  Hello, world!
</main>
`;
